homeassistant/components/modbus/sensor.py

```diff
@@ -250,16 +250,32 @@ def update(self):
             registers.reverse()
 
         byte_string = b"".join([x.to_bytes(2, byteorder="big") for x in registers])
-        if self._data_type != DATA_TYPE_STRING:
-            val = struct.unpack(self._structure, byte_string)[0]
-            val = self._scale * val + self._offset
-            if isinstance(val, int):
-                self._value = str(val)
-                if self._precision > 0:
-                    self._value += "." + "0" * self._precision
-            else:
-                self._value = f"{val:.{self._precision}f}"
-        else:
+        if self._data_type == DATA_TYPE_STRING:
             self._value = byte_string.decode()
+        else:
+            val = struct.unpack(self._structure, byte_string)
+
+            # Issue: https://github.com/home-assistant/core/issues/41944
+            # If unpack() returns a tuple greater than 1, don't try to process the value.
+            # Instead, return the values of unpack(...) separated by commas.
+            if len(val) > 1:
+                self._value = ",".join(map(str, val))
+            else:
+                val = val[0]
```

Comment:
join works it out automatically if there is 1 item.

Should it be `", "`
```suggestion
            self._value = ",".join(map(str, val))
```

Reply:
I would prefer without space to avoid subsequent strip() calls when extracting values. A template sensor will be most likely used to create multiple sensor out of one read.

Reply:
I agree avoiding a not needed space is preferable.

Reply:
I will update as you suggested.

Reply:
I also think the space is redundant in this case. If you all agree, I will keep it as it is.

Reply:
The current method of no spaces has been working perfectly since I adopted the code changes on the 25th Oct.
It will just result in more stripping of data.